test/cache.py

```diff
@@ -110,3 +114,22 @@ def test_corrupt_gz_cache(self):
         f.close()
         FileCache(filename)
         shutil.rmtree(tempdir)
+
+    @mock.patch('vsc.utils.cache.jsonpickle.decode')
+    def test_value_error(self, mock_decode):
+        "Test to see that a ValueError upon decoding gets caught correctly"
+        tempdir = tempfile.mkdtemp()
+        # create a tempfilename
+        (handle, filename) = tempfile.mkstemp(dir=tempdir)
+        f = os.fdopen(handle, 'w')
+        g = gzip.GzipFile(mode='w', fileobj=f)
+        g.write('blabla no json gzip stuffz')
+        g.close()
+
+        e = ValueError('unable to find valid JSON')
+        mock_decode.side_effect = e
```

Comment:
why mock this side effect if you're first going trought the trouble of creating a file that will generate this error in the first place?
Can't you just remove the mocking?